src/android/device.rs

```diff
@@ -40,15 +108,72 @@ impl Reportable for ApkBuildError {
     }
 }
 
+#[derive(Debug)]
+pub enum AabBuildError {
+    BuildFailed(bossy::Error),
+}
+
+impl Reportable for AabBuildError {
+    fn report(&self) -> Report {
+        match self {
+            Self::BuildFailed(err) => Report::error("Failed to build AAB", err),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub enum ApksBuildError {
+    CleanFailed(bossy::Error),
+    BuildFromAabFailed(bossy::Error),
+}
+
+impl Reportable for ApksBuildError {
+    fn report(&self) -> Report {
+        match self {
+            Self::CleanFailed(err) => Report::error("Failed to clean old APKS", err),
+            Self::BuildFromAabFailed(err) => Report::error("Failed to build APKS from AAB", err),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub enum BundletoolInstallError {
+    #[cfg(target_os = "macos")]
+    MacOSInstallFailed(crate::apple::deps::Error),
+    DownloadFailed(ureq::Error),
+    JarFileCreationFailed(std::io::Error),
+    CopyToFileFailed(std::io::Error),
+    MoveFileFailed(std::io::Error),
+}
+
+impl Reportable for BundletoolInstallError {
+    fn report(&self) -> Report {
+        match self {
+            #[cfg(target_os = "macos")]
+            Self::MacOSInstallFailed(err) => Report::error("Failed to install `bundletool`", err),
+            Self::DownloadFailed(err) => Report::error("Failed to download `bundletool`", err),
+            Self::JarFileCreationFailed(err) => {
```

Comment:
fs-related errors should also include the path (search `{ path, cause }` for existing patterns)